packages/financial-templates-lib/src/price-feed/LPPriceFeed.js

```diff
@@ -0,0 +1,129 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+const { BlockFinder } = require("./utils");
+const { ConvertDecimals } = require("@uma/common");
+const assert = require("assert");
+class LPPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs new price feed object that tracks how much of a provided token a single LP share is redeemable
+   *         for.
+   * @dev Note: this can support most LP shares and may support other types of pool contracts.
+   * @param {Object} logger Winston module used to send logs.
+   * @param {Object} web3 Provider from Truffle instance to connect to Ethereum network.
+   * @param {Object} erc20Abi ERC20 abi object to create a contract instance.
+   * @param {String} poolAddress Ethereum address of the LP pool to monitor.
+   * @param {String} tokenAddress Ethereum address of the per-share token balance we're tracking within the pool.
+   * @param {Function} getTime Returns the current time.
+   * @param {Function} [blockFinder] Optionally pass in a shared blockFinder instance (to share the cache).
+   * @param {Integer} [minTimeBetweenUpdates] Minimum amount of time that must pass before update will actually run
+   *                                        again.
+   * @param {Integer} [priceFeedDecimals] Precision that the caller wants precision to be reported in.
+   * @return None or throws an Error.
+   */
+  constructor({
+    logger,
+    web3,
+    erc20Abi,
+    poolAddress,
+    tokenAddress,
+    getTime,
+    blockFinder,
+    minTimeBetweenUpdates = 60,
+    priceFeedDecimals = 18
+  }) {
+    super();
+
+    // Assert required arguments.
+    assert(logger, "logger required");
```

Comment:
Are these asserts needed for any fields that are in the requiredFields dict in CreatePriceFeed?

Reply:
I guess we _could_ leave them out, but I was thinking of them as a way to fail early if the CreatePriceFeed implementation was wrong. Thoughts?